channeld/channeld.c

```diff
@@ -2915,6 +2915,44 @@ static size_t calc_weight(enum tx_role role, const struct wally_psbt *psbt)
 	return weight;
 }
 
+/* Get the remote node amount after the splice */
+static struct amount_msat remote_balance(struct peer *peer,
+					    enum tx_role our_role,
+					    const struct wally_psbt *psbt,
+					    int chan_output_index,
+					    int chan_input_index)
+{
+	s64 relative_remote_val;
+	/* Current value of the remote channel balance */
+	struct amount_msat push_value;
+	const bool opener = our_role == TX_INITIATOR;
+
+	/* We calculcate the `push_value` to send to the
+	 * hsmd, that is the remote amount in the channel
+	 * after the splice. */
+	switch (our_role) {
```

Comment:
this is hard to analyze, because `our_role` is used to calculate `opener`, but then both `opener` and `our_role` are used in the logic below.  it would be best to use only one or the other.